### PR TITLE
Merge dev2

### PR DESCRIPTION
**Pull request by MichaelDvP**
only a few manual edits nedded to merge, mainly mqtt.
It compiles, but i havn't tested.
The standalone build still needs some changes for the espmqttclient.

### CODE REVIEW

File: interface/src/project/DashboardDevicesDialog.tsx
Comment (proddy) [inline interface/src/project/DashboardDevicesDialog.tsx:1]: I optimized this dialog code when doing my alova port so need to check whether these changes are still valid.

Comment (proddy) [reply interface/src/project/DashboardDevicesDialog.tsx:1]: I'll merge and check afterwards

Comment (MichaelDvP) [reply interface/src/project/DashboardDevicesDialog.tsx:1]: This is the change for number input of custom entities without UOM. It does not conflict with your changes.

Comment (proddy) [reply interface/src/project/DashboardDevicesDialog.tsx:1]: ok! cool.

**Pull request merged**